sql/threadpool_unix.cc

```diff
@@ -1304,7 +1304,7 @@ void tp_wait_end(THD *thd) {
 static void set_next_timeout_check(ulonglong abstime) {
   DBUG_ENTER("set_next_timeout_check");
   while (abstime < pool_timer.next_timeout_check.load()) {
-    ulong old = pool_timer.next_timeout_check.load();
+    ulonglong old = pool_timer.next_timeout_check.load();
```

Comment:
`next_timeout_check` is declared as
```
  std::atomic<uint64> next_timeout_check;
```
So, may be 
```suggestion
    uint64 old = pool_timer.next_timeout_check.load();
```
or even `auto`